Multiple failures in Model Tests

### Area

testsuite

### Describe the bug

Model Test failed in all 4 runs last night:
https://github.com/keycloak/keycloak/actions/workflows/ci.yml?query=event%3Aschedule+branch%3Amain

Haven't looked at all logs, but looks like it may be caused by:

https://github.com/keycloak/keycloak/commit/168734b817da2138548bacba973204e80f566b2c

Based on the following log:

```
2022-12-02T02:54:26.0363842Z [ERROR] Tests run: 9, Failures: 1, Errors: 7, Skipped: 1, Time elapsed: 6.118 s <<< FAILURE! - in org.keycloak.testsuite.model.session.OfflineSessionPersistenceTest
2022-12-02T02:54:26.0369472Z [ERROR] org.keycloak.testsuite.model.session.OfflineSessionPersistenceTest.testPersistenceMultipleNodesClientSessionsAtRandomNode  Time elapsed: 0.316 s  <<< ERROR!
2022-12-02T02:54:26.0370627Z java.lang.NullPointerException
2022-12-02T02:54:26.0371431Z 	at org.keycloak.services.HttpRequestImpl.getHttpHeaders(HttpRequestImpl.java:46)
2022-12-02T02:54:26.0372383Z 	at org.keycloak.services.DefaultKeycloakContext.getRequestHeaders(DefaultKeycloakContext.java:98)
2022-12-02T02:54:26.0373407Z 	at org.keycloak.device.DeviceActivityManager.getDeviceFromUserAgent(DeviceActivityManager.java:89)
2022-12-02T02:54:26.0374379Z 	at org.keycloak.device.DeviceActivityManager.attachDevice(DeviceActivityManager.java:75)
2022-12-02T02:54:26.0375457Z 	at org.keycloak.models.sessions.infinispan.InfinispanUserSessionProvider.createUserSession(InfinispanUserSessionProvider.java:236)
2022-12-02T02:54:26.0376632Z 	at org.keycloak.models.sessions.infinispan.InfinispanUserSessionProvider.createUserSession(InfinispanUserSessionProvider.java:219)
2022-12-02T02:54:26.0377800Z 	at org.keycloak.testsuite.model.session.OfflineSessionPersistenceTest.createOfflineSession(OfflineSessionPersistenceTest.java:422)
```



### Version

main

### Expected behavior

All tests pass

### Actual behavior

Tests fails 100% time

### How to Reproduce?

See logs from:
https://github.com/keycloak/keycloak/actions/workflows/ci.yml?query=event%3Aschedule+branch%3Amain

### Anything else?

_No response_

--- a/services/src/main/java/org/keycloak/services/HttpRequestImpl.java
+++ b/services/src/main/java/org/keycloak/services/HttpRequestImpl.java
@@ -17,42 +17,58 @@
 
 package org.keycloak.services;
 
-import java.security.cert.X509Certificate;
+import org.keycloak.http.HttpRequest;
+
 import javax.ws.rs.core.HttpHeaders;
 import javax.ws.rs.core.MultivaluedMap;
 import javax.ws.rs.core.UriInfo;
-import org.keycloak.http.HttpRequest;
+import java.security.cert.X509Certificate;
 
 public class HttpRequestImpl implements HttpRequest {
 
     private org.jboss.resteasy.spi.HttpRequest delegate;
 
-    public <R> HttpRequestImpl(org.jboss.resteasy.spi.HttpRequest delegate) {
+    public HttpRequestImpl(org.jboss.resteasy.spi.HttpRequest delegate) {
         this.delegate = delegate;
     }
 
     @Override
     public String getHttpMethod() {
+        if (delegate == null) {
+            return null;
+        }
         return delegate.getHttpMethod();
     }
 
     @Override
     public MultivaluedMap<String, String> getDecodedFormParameters() {
+        if (delegate == null) {
+            return null;
+        }
         return delegate.getDecodedFormParameters();
     }
 
     @Override
     public HttpHeaders getHttpHeaders() {
+        if (delegate == null) {
+            return null;
+        }
         return delegate.getHttpHeaders();
     }
 
     @Override
     public X509Certificate[] getClientCertificateChain() {
+        if (delegate == null) {
+            return null;
+        }
         return (X509Certificate[]) delegate.getAttribute("javax.servlet.request.X509Certificate");
     }
 
     @Override
     public UriInfo getUri() {
+        if (delegate == null) {
+            return null;
+        }
         return delegate.getUri();
     }
 }

